dot-object/dot-object.d.ts

```diff
@@ -0,0 +1,162 @@
+// Type definitions for Dot-Object v1.4.1
+// Project: https://github.com/rhalff/dot-object
+// Definitions by: Niko Kovačič <https://github.com/nkovacic>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
+declare namespace DotObject {
+    interface Dot {
+        new(separator: string): Dot;
+        /**
+         *
+         * Copy a property from one object to another object.
+         *
+         * If the source path does not exist (undefined)
+         * the property on the other object will not be set.
+         *
+         * @param {String} source
+         * @param {String} target
+         * @param {Object} obj1
+         * @param {Object} obj2
+         * @param {Function|Array} mods
+         * @param {Boolean} merge
+         */
+        copy(source: string, target: string, obj1: any, obj2: any, mods?: Function | Array<Function>, merge?: boolean): void;
+        /**
+         *
+         * Convert object to dotted-key/value pair
+         *
+         * Usage:
+         *
+         *   var tgt = dot.dot(obj)
+         *
+         *   or
+         *
+         *   var tgt = {}
+         *   dot.dot(obj, tgt)
+         *
+         * @param {Object} obj source object
+         * @param {Object} tgt target object
+         */
+        dot(obj: any, tgt: any): void
+        /**
+         *
+         * Remove value from an object using dot notation.
+         *
+         * @param {String} path
+         * @param {Object} obj
+         * @return {Mixed} The removed value
+         */
+        del(path: string, obj: any): any;
+        /**
+         *
+         * Move a property from one place to the other.
+         *
+         * If the source path does not exist (undefined)
+         * the target property will not be set.
+         *
+         * @param {String} source
+         * @param {String} target
+         * @param {Object} obj
+         * @param {Function|Array} mods
+         * @param {Boolean} merge
+         */
+        move(source: string, target: string, obj: any, mods?: Function | Array<Function>, merge?: boolean): void;
```

Comment:
Everywhere in this file should use a more specific function type instead of `Function`, probably `(arg: any) => any`